File: services/user-service/src/main/java/info/novatec/spring/showcase/user/repository/UserRepository.java
package info.novatec.spring.showcase.user.repository;

import info.novatec.spring.showcase.user.model.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Long> {

  Optional<User> findOneByIdpId(String idpId);
}

File: services/like-service/src/main/kotlin/info/novatec/repository/UserRepository.kt
package info.novatec.repository

import info.novatec.model.User
import io.micronaut.data.annotation.Id
import io.micronaut.data.jdbc.annotation.JdbcRepository
import io.micronaut.data.model.query.builder.sql.Dialect
import io.micronaut.data.repository.CrudRepository
import java.util.*

@JdbcRepository(dialect = Dialect.MYSQL)
interface UserRepository : CrudRepository<User, UUID> {
    fun update(@Id id: UUID, name: String)
}